src/json.c

```diff
@@ -178,6 +214,40 @@ int JSON_StoreVectorInDocField(FieldSpec *fs, JSONResultsIterator arrIter, struc
   return REDISMODULE_OK;
 }
 
+int JSON_StoreTextInDocField(size_t len, JSONResultsIterator jsonIter, struct DocumentField *df) {
+  df->multiVal = rm_calloc(len , sizeof(*df->multiVal));
+  
+  int i = 0, nulls = 0;
+  size_t strlen;
+  RedisJSON json;
+  const char *str;
+  while ((json = japi->next(jsonIter))) {
+    JSONType jsonType = japi->getType(json);
+    if (jsonType == JSONType_String) {
+      japi->getString(json, &str, &strlen);
+      df->multiVal[i++] = rm_strndup(str, strlen);
```

Comment:
```suggestion
      df->multiVal[i++] = rm_strndup(str, strlen(str));
```
I wonder how it works...

Reply:
Ohh ok, you named you variable this way :+1: